gammapy/utils/fitting/fit.py

```diff
@@ -55,88 +67,41 @@ def total_stat(self, parameters):
-        return profiles
-
-    def likelihood_profile(self, model, parameter, values=None, bounds=2, nvalues=11):
-        """Compute likelihood profile for a single parameter of the model.
+        Run all fitting steps.
 
         Parameters
         ----------
-        model : `~gammapy.spectrum.models.SpectralModel`
-            Model to compute the likelihood profile for.
-        parameter : str
-            Parameter to calculate profile for
-        values : `~astropy.units.Quantity` (optional)
-            Parameter values to evaluate the likelihood for.
-        bounds : int or tuple of float
-            When an `int` is passed the bounds are computed from `bounds * sigma`
-            from the best fit value of the parameter, where `sigma` corresponds to
-            the one sigma error on the parameter. If a tuple of floats is given
-            those are taken as the min and max values and `nvalues` are linearly
-            spaced between those.
-        nvalues : int
-            Number of parameter grid points to use.
+        optimize_opts : dict
+            Options passed to `Fit.optimize`.
+        covar_opts : dict
+            Options passed to `Fit.covar`.
 
         Returns
         -------
-        likelihood_profile : dict
-            Dict of parameter values and likelihood values.
-        """
-        self._model = model.copy()
-
-        likelihood = []
-
-        if values is None:
-            if isinstance(bounds, tuple):
-                parmin, parmax = bounds
-            else:
-                parerr = model.parameters.error(parameter)
-                parval = model.parameters[parameter].value
-                parmin, parmax = parval - bounds * parerr, parval + bounds * parerr
-
-            values = np.linspace(parmin, parmax, nvalues)
-
-        for value in values:
-            self._model.parameters[parameter].value = value
-            stat = self.total_stat(self._model.parameters)
-            likelihood.append(stat)
-
-        return {"values": values, "likelihood": np.array(likelihood)}
-
-    def sqrt_ts(self, parameters):
-        """Compute the sqrt(TS) of a model against the null hypthesis, that
-        the amplitude of the model is zero.
+        fit_result : `FitResult`
+            Results
         """
-        stat_best_fit = self.total_stat(parameters)
+        if optimize_opts is None:
+            optimize_opts = {}
+        optimize_result = self.optimize(**optimize_opts)
 
-        # store best fit amplitude, set amplitude of fit model to zero
-        amplitude = parameters["amplitude"].value
-        parameters["amplitude"].value = 0
-        stat_null = self.total_stat(parameters)
+        if covar_opts is None:
+            covar_opts = {}
 
-        # set amplitude of fit model to best fit amplitude
-        parameters["amplitude"].value = amplitude
+        covar_opts.setdefault("backend", "minuit")
+        if covar_opts["backend"] in registry.register["covar"]:
```

Comment:
I think this logic is at the wrong place, no? I think it should mainly reduce to the one line `covar_result = self.covar(**covar_opts)`, the rest should be handled either in `Fit.covar` or the `Registry.` (like raising an error if a backend does not support covar). There is no need to set "covar_opts.setdefault("backend", "minuit")" either, as it is already set as a default in `Fit.covar`.

Reply:
The registry always gives an error. Here I want to emit a warning and continue.
I've changed this a bit, using a simpler `if` clause, which I prefer over try-except in this case.